modules/table/src/main/java/org/apache/ignite/internal/table/distributed/raft/snapshot/PartitionAccessImpl.java

```diff
@@ -57,48 +70,147 @@ public PartitionKey partitionKey() {
+    }
 
-        // TODO: IGNITE-18030 - actually recreate or do in a different way
-        //return mvTableStorage.destroyPartition(partId())
-        return CompletableFuture.completedFuture(null)
-                .thenApply(unused -> mvTableStorage.getOrCreateMvPartition(partId()));
+    @Override
+    public void addTxMeta(UUID txId, TxMeta txMeta) {
+        getTxStateStorage(partitionId()).put(txId, txMeta);
     }
 
     @Override
-    public TxStateStorage reCreateTxStatePartitionStorage() throws StorageException {
-        assert txStateTableStorage.getTxStateStorage(partId()) != null : "table=" + tableName() + ", part=" + partId();
+    public @Nullable RowId closestRowId(RowId lowerBound) {
+        return getMvPartitionStorage(partitionId()).closestRowId(lowerBound);
+    }
 
-        // TODO: IGNITE-18030 - actually recreate or do in a different way
-        //txStateTableStorage.destroyTxStateStorage(partId());
+    @Override
+    public Cursor<ReadResult> getAllRowVersions(RowId rowId) {
+        return getMvPartitionStorage(partitionId()).scanVersions(rowId);
+    }
 
-        return txStateTableStorage.getOrCreateTxStateStorage(partId());
+    @Override
+    public @Nullable RaftGroupConfiguration committedGroupConfiguration() {
+        return getMvPartitionStorage(partitionId()).committedGroupConfiguration();
     }
 
-    private int partId() {
-        return partitionKey.partitionId();
+    @Override
+    public void addWrite(RowId rowId, TableRow row, UUID txId, UUID commitTableId, int commitPartitionId) {
+        MvPartitionStorage mvPartitionStorage = getMvPartitionStorage(partitionId());
+
+        mvPartitionStorage.runConsistently(() -> mvPartitionStorage.addWrite(rowId, row, txId, commitTableId, commitPartitionId));
     }
 
-    private String tableName() {
-        return mvTableStorage.configuration().name().value();
+    @Override
+    public void addWriteCommitted(RowId rowId, TableRow row, HybridTimestamp commitTimestamp) {
+        MvPartitionStorage mvPartitionStorage = getMvPartitionStorage(partitionId());
+
+        mvPartitionStorage.runConsistently(() -> {
+            mvPartitionStorage.addWriteCommitted(rowId, row, commitTimestamp);
+
+            return null;
+        });
+    }
+
+    @Override
+    public void updateLastApplied(long lastAppliedIndex, long lastAppliedTerm, RaftGroupConfiguration raftGroupConfig) {
+        MvPartitionStorage mvPartitionStorage = getMvPartitionStorage(partitionId());
+        TxStateStorage txStateStorage = getTxStateStorage(partitionId());
+
+        mvPartitionStorage.runConsistently(() -> {
+            mvPartitionStorage.lastApplied(lastAppliedIndex, lastAppliedTerm);
+
+            txStateStorage.lastApplied(lastAppliedIndex, lastAppliedTerm);
+
+            mvPartitionStorage.committedGroupConfiguration(raftGroupConfig);
+
+            return null;
+        });
+    }
+
+    @Override
+    public long minLastAppliedIndex() {
```

Comment:
Please add unit tests for all 4 min/max methods. This logic was implemented in other places before this change, and it was covered by tests (as it seems really important). So we probably need these tests here now.

Reply:
I think there are already tests:
- `org.apache.ignite.internal.table.distributed.raft.snapshot.PartitionSnapshotStorageFactoryTest#testForChoosingMinimumAppliedIndexForMeta`
- `org.apache.ignite.internal.table.distributed.raft.snapshot.outgoing.OutgoingSnapshotReaderTest#testForChoosingMaximumAppliedIndexForMeta`

Reply:
Both tests that you mention make sure that `PartitionAccess#min/max...` methods get called. But here, we need to make that that these methods actually work correctly (meaning, that they actually calculate min or max of the corresponding values). This is not tested anywhere.

Reply:
Understood you, I will add tests.